host.py

```diff
@@ -1,24 +1,85 @@
 import json
+import logging
 import os
 import shlex
 import subprocess
 import sys
+import threading
 import typing
 
 from abc import ABC
 from abc import abstractmethod
+from collections.abc import Iterable
+from collections.abc import Mapping
 from dataclasses import dataclass
 from typing import Any
+from typing import Optional
 
 from logger import logger
 
 
+_lock = threading.Lock()
+
+_global_counter_val = 0
+
+
+def _global_counter() -> int:
+    # For each run() call, we log a message when staring the command and when
```

Comment:
Typo for "staring" -> "starting"
Maybe a more descriptive name than _global_counter would be better such as "_unique_log_id"? WDYT?